In [1]:
import pandas as pd
import numpy as ny

In [2]:
dff = pd.read_csv("dataset/fake.csv")
dff

title  \
0       Donald Trump Sends Out Embarrassing New Year’...   
1       Drunk Bragging Trump Staffer Started Russian ...   
2       Sheriff David Clarke Becomes An Internet Joke...   
3       Trump Is So Obsessed He Even Has Obama’s Name...   
4       Pope Francis Just Called Out Donald Trump Dur...   
...                                                  ...   
23476  McPain: John McCain Furious That Iran Treated ...   
23477  JUSTICE? Yahoo Settles E-mail Privacy Class-ac...   
23478  Sunnistan: US and Allied ‘Safe Zone’ Plan to T...   
23479  How to Blow $700 Million: Al Jazeera America F...   
23480  10 U.S. Navy Sailors Held by Iranian Military ...   

                                                    text      subject  \
0      Donald Trump just couldn t wish all Americans ...         News   
1      House Intelligence Committee Chairman Devin Nu...         News   
2      On Friday, it was revealed that former Milwauk...         News   
3      On Christmas day, Donald Trump announced that ...         News   
4      Pope Francis used his annual Christmas Day mes...         News   
...                                                  ...          ...   
23476  21st Century Wire says As 21WIRE reported earl...  Middle-east   
23477  21st Century Wire says It s a familiar theme. ...  Middle-east   
23478  Patrick Henningsen  21st Century WireRemember ...  Middle-east   
23479  21st Century Wire says Al Jazeera America will...  Middle-east   
23480  21st Century Wire says As 21WIRE predicted in ...  Middle-east   

                    date  
0      December 31, 2017  
1      December 31, 2017  
2      December 30, 2017  
3      December 29, 2017  
4      December 25, 2017  
...                  ...  
23476   January 16, 2016  
23477   January 16, 2016  
23478   January 15, 2016  
23479   January 14, 2016  
23480   January 12, 2016  

[23481 rows x 4 columns]

In [3]:
print(dff.isnull().sum())

title      0
text       0
subject    0
date       0
dtype: int64


In [4]:
def checkDuplicated(df):
    duplicated_count = df.duplicated().sum()

    if duplicated_count > 0:
        print("Duplicated rows:")
        print(df[df.duplicated()])

        
        df = df.drop_duplicates()
    else:
        print("There are no duplicated rows.")

    return df

checkDuplicated(dff)

Duplicated rows:
                                                   title  \
9942   HILLARY TWEETS MESSAGE In Defense Of DACA…OOPS...   
11446  FORMER DEMOCRAT WARNS Young Americans: “Rioter...   
14925  [VIDEO] #BlackLivesMatter Terrorists Storm Dar...   

                                                    text   subject  \
9942   No time to waste   we've got to fight with eve...  politics   
11446   Who is silencing political speech, physically...  politics   
14925  They were probably just looking for a  safe sp...  politics   

               date  
9942    Sep 9, 2017  
11446  Mar 10, 2017  
14925  Nov 16, 2015  


title  \
0       Donald Trump Sends Out Embarrassing New Year’...   
1       Drunk Bragging Trump Staffer Started Russian ...   
2       Sheriff David Clarke Becomes An Internet Joke...   
3       Trump Is So Obsessed He Even Has Obama’s Name...   
4       Pope Francis Just Called Out Donald Trump Dur...   
...                                                  ...   
23476  McPain: John McCain Furious That Iran Treated ...   
23477  JUSTICE? Yahoo Settles E-mail Privacy Class-ac...   
23478  Sunnistan: US and Allied ‘Safe Zone’ Plan to T...   
23479  How to Blow $700 Million: Al Jazeera America F...   
23480  10 U.S. Navy Sailors Held by Iranian Military ...   

                                                    text      subject  \
0      Donald Trump just couldn t wish all Americans ...         News   
1      House Intelligence Committee Chairman Devin Nu...         News   
2      On Friday, it was revealed that former Milwauk...         News   
3      On Christmas day, Donald Trump announced that ...         News   
4      Pope Francis used his annual Christmas Day mes...         News   
...                                                  ...          ...   
23476  21st Century Wire says As 21WIRE reported earl...  Middle-east   
23477  21st Century Wire says It s a familiar theme. ...  Middle-east   
23478  Patrick Henningsen  21st Century WireRemember ...  Middle-east   
23479  21st Century Wire says Al Jazeera America will...  Middle-east   
23480  21st Century Wire says As 21WIRE predicted in ...  Middle-east   

                    date  
0      December 31, 2017  
1      December 31, 2017  
2      December 30, 2017  
3      December 29, 2017  
4      December 25, 2017  
...                  ...  
23476   January 16, 2016  
23477   January 16, 2016  
23478   January 15, 2016  
23479   January 14, 2016  
23480   January 12, 2016  

[23478 rows x 4 columns]

In [5]:
dft = pd.read_csv("dataset/true.csv")
dft

title  \
0      As U.S. budget fight looms, Republicans flip t...   
1      U.S. military to accept transgender recruits o...   
2      Senior U.S. Republican senator: 'Let Mr. Muell...   
3      FBI Russia probe helped by Australian diplomat...   
4      Trump wants Postal Service to charge 'much mor...   
...                                                  ...   
21412  'Fully committed' NATO backs new U.S. approach...   
21413  LexisNexis withdrew two products from Chinese ...   
21414  Minsk cultural hub becomes haven from authorities   
21415  Vatican upbeat on possibility of Pope Francis ...   
21416  Indonesia to buy $1.14 billion worth of Russia...   

                                                    text       subject  \
0      WASHINGTON (Reuters) - The head of a conservat...  politicsNews   
1      WASHINGTON (Reuters) - Transgender people will...  politicsNews   
2      WASHINGTON (Reuters) - The special counsel inv...  politicsNews   
3      WASHINGTON (Reuters) - Trump campaign adviser ...  politicsNews   
4      SEATTLE/WASHINGTON (Reuters) - President Donal...  politicsNews   
...                                                  ...           ...   
21412  BRUSSELS (Reuters) - NATO allies on Tuesday we...     worldnews   
21413  LONDON (Reuters) - LexisNexis, a provider of l...     worldnews   
21414  MINSK (Reuters) - In the shadow of disused Sov...     worldnews   
21415  MOSCOW (Reuters) - Vatican Secretary of State ...     worldnews   
21416  JAKARTA (Reuters) - Indonesia will buy 11 Sukh...     worldnews   

                     date  
0      December 31, 2017   
1      December 29, 2017   
2      December 31, 2017   
3      December 30, 2017   
4      December 29, 2017   
...                   ...  
21412    August 22, 2017   
21413    August 22, 2017   
21414    August 22, 2017   
21415    August 22, 2017   
21416    August 22, 2017   

[21417 rows x 4 columns]

In [6]:
print(dft.isnull().sum())

title      0
text       0
subject    0
date       0
dtype: int64


In [7]:
checkDuplicated(dft)

Duplicated rows:
                                                   title  \
445    Senate tax bill stalls on deficit-focused 'tri...   
778    Trump warns 'rogue regime' North Korea of grav...   
892    Republicans unveil tax cut bill, but the hard ...   
896    Trump taps Fed centrist Powell to lead U.S. ce...   
974    Two ex-Trump aides charged in Russia probe, th...   
...                                                  ...   
21228  France unveils labor reforms in first step to ...   
21263  Guatemala top court sides with U.N. graft unit...   
21290  Europeans, Africans agree renewed push to tack...   
21353  Thailand's ousted PM Yingluck has fled abroad:...   
21408  U.S., North Korea clash at U.N. forum over nuc...   

                                                    text       subject  \
445    WASHINGTON (Reuters) - The U.S. Senate on Thur...  politicsNews   
778    BEIJING (Reuters) - U.S. President Donald Trum...  politicsNews   
892    WASHINGTON (Reuters) - U.S. House

title  \
0      As U.S. budget fight looms, Republicans flip t...   
1      U.S. military to accept transgender recruits o...   
2      Senior U.S. Republican senator: 'Let Mr. Muell...   
3      FBI Russia probe helped by Australian diplomat...   
4      Trump wants Postal Service to charge 'much mor...   
...                                                  ...   
21412  'Fully committed' NATO backs new U.S. approach...   
21413  LexisNexis withdrew two products from Chinese ...   
21414  Minsk cultural hub becomes haven from authorities   
21415  Vatican upbeat on possibility of Pope Francis ...   
21416  Indonesia to buy $1.14 billion worth of Russia...   

                                                    text       subject  \
0      WASHINGTON (Reuters) - The head of a conservat...  politicsNews   
1      WASHINGTON (Reuters) - Transgender people will...  politicsNews   
2      WASHINGTON (Reuters) - The special counsel inv...  politicsNews   
3      WASHINGTON (Reuters) - Trump campaign adviser ...  politicsNews   
4      SEATTLE/WASHINGTON (Reuters) - President Donal...  politicsNews   
...                                                  ...           ...   
21412  BRUSSELS (Reuters) - NATO allies on Tuesday we...     worldnews   
21413  LONDON (Reuters) - LexisNexis, a provider of l...     worldnews   
21414  MINSK (Reuters) - In the shadow of disused Sov...     worldnews   
21415  MOSCOW (Reuters) - Vatican Secretary of State ...     worldnews   
21416  JAKARTA (Reuters) - Indonesia will buy 11 Sukh...     worldnews   

                     date  
0      December 31, 2017   
1      December 29, 2017   
2      December 31, 2017   
3      December 30, 2017   
4      December 29, 2017   
...                   ...  
21412    August 22, 2017   
21413    August 22, 2017   
21414    August 22, 2017   
21415    August 22, 2017   
21416    August 22, 2017   

[21211 rows x 4 columns]

In [8]:
dft = checkDuplicated(dft)
dff = checkDuplicated(dff)

Duplicated rows:
                                                   title  \
445    Senate tax bill stalls on deficit-focused 'tri...   
778    Trump warns 'rogue regime' North Korea of grav...   
892    Republicans unveil tax cut bill, but the hard ...   
896    Trump taps Fed centrist Powell to lead U.S. ce...   
974    Two ex-Trump aides charged in Russia probe, th...   
...                                                  ...   
21228  France unveils labor reforms in first step to ...   
21263  Guatemala top court sides with U.N. graft unit...   
21290  Europeans, Africans agree renewed push to tack...   
21353  Thailand's ousted PM Yingluck has fled abroad:...   
21408  U.S., North Korea clash at U.N. forum over nuc...   

                                                    text       subject  \
445    WASHINGTON (Reuters) - The U.S. Senate on Thur...  politicsNews   
778    BEIJING (Reuters) - U.S. President Donald Trum...  politicsNews   
892    WASHINGTON (Reuters) - U.S. House

In [9]:
rows1, cols1 = dff.shape
rows2, cols2 = dft.shape

In [10]:
print(rows1)
print(rows2)

23478
21211


## i have in the true news 21211 rows
##  in the false news 23478 rows

In [11]:
dff1 = dff.iloc[:rows2]
dff2 = dff.iloc[rows2:]  

In [12]:
print(dff1.shape)
print(dff2.shape)

(21211, 4)
(2267, 4)


In [13]:
print(dft.shape)

(21211, 4)


In [14]:
dft["y"] = 1

C:\Users\DELL\AppData\Local\Temp\ipykernel_18208\4201324816.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dft["y"] = 1


In [15]:
dft

title  \
0      As U.S. budget fight looms, Republicans flip t...   
1      U.S. military to accept transgender recruits o...   
2      Senior U.S. Republican senator: 'Let Mr. Muell...   
3      FBI Russia probe helped by Australian diplomat...   
4      Trump wants Postal Service to charge 'much mor...   
...                                                  ...   
21412  'Fully committed' NATO backs new U.S. approach...   
21413  LexisNexis withdrew two products from Chinese ...   
21414  Minsk cultural hub becomes haven from authorities   
21415  Vatican upbeat on possibility of Pope Francis ...   
21416  Indonesia to buy $1.14 billion worth of Russia...   

                                                    text       subject  \
0      WASHINGTON (Reuters) - The head of a conservat...  politicsNews   
1      WASHINGTON (Reuters) - Transgender people will...  politicsNews   
2      WASHINGTON (Reuters) - The special counsel inv...  politicsNews   
3      WASHINGTON (Reuters) - Trump campaign adviser ...  politicsNews   
4      SEATTLE/WASHINGTON (Reuters) - President Donal...  politicsNews   
...                                                  ...           ...   
21412  BRUSSELS (Reuters) - NATO allies on Tuesday we...     worldnews   
21413  LONDON (Reuters) - LexisNexis, a provider of l...     worldnews   
21414  MINSK (Reuters) - In the shadow of disused Sov...     worldnews   
21415  MOSCOW (Reuters) - Vatican Secretary of State ...     worldnews   
21416  JAKARTA (Reuters) - Indonesia will buy 11 Sukh...     worldnews   

                     date  y  
0      December 31, 2017   1  
1      December 29, 2017   1  
2      December 31, 2017   1  
3      December 30, 2017   1  
4      December 29, 2017   1  
...                   ... ..  
21412    August 22, 2017   1  
21413    August 22, 2017   1  
21414    August 22, 2017   1  
21415    August 22, 2017   1  
21416    August 22, 2017   1  

[21211 rows x 5 columns]

In [16]:
dff1["y"] = 0
dff2["y"] = 0

C:\Users\DELL\AppData\Local\Temp\ipykernel_18208\1920582133.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dff1["y"] = 0
C:\Users\DELL\AppData\Local\Temp\ipykernel_18208\1920582133.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dff2["y"] = 0


In [17]:
dff1

title  \
0       Donald Trump Sends Out Embarrassing New Year’...   
1       Drunk Bragging Trump Staffer Started Russian ...   
2       Sheriff David Clarke Becomes An Internet Joke...   
3       Trump Is So Obsessed He Even Has Obama’s Name...   
4       Pope Francis Just Called Out Donald Trump Dur...   
...                                                  ...   
21209  ONLY IN DETROIT: Entitled Squatter Gets Squatt...   
21210  FOX NEWS REPORTER ASKS MAYOR Why She’s Using T...   
21211  NEW ACCUSATION AGAINST BERNIE SANDERS Exposes ...   
21212  JESUS CHRIST IS STRIPPED From Christmas Celebr...   
21213  Michael Moore Wants America To Know “We Are Al...   

                                                    text    subject  \
0      Donald Trump just couldn t wish all Americans ...       News   
1      House Intelligence Committee Chairman Devin Nu...       News   
2      On Friday, it was revealed that former Milwauk...       News   
3      On Christmas day, Donald Trump announced that ...       News   
4      Pope Francis used his annual Christmas Day mes...       News   
...                                                  ...        ...   
21209                   When karma bites ya  in the a*s   left-news   
21210  The drip drip drip of the Left ensuring Christ...  left-news   
21211  By now, it s pretty clear that Bernie doesn t ...  left-news   
21212  Does that mean Islam is not considered a faith...  left-news   
21213  No dumb ass  we are actually NOT all Muslims! ...  left-news   

                    date  y  
0      December 31, 2017  0  
1      December 31, 2017  0  
2      December 30, 2017  0  
3      December 29, 2017  0  
4      December 25, 2017  0  
...                  ... ..  
21209       Dec 18, 2015  0  
21210       Dec 18, 2015  0  
21211       Dec 18, 2015  0  
21212       Dec 18, 2015  0  
21213       Dec 17, 2015  0  

[21211 rows x 5 columns]

In [18]:
dff2

title  \
21214  PAUL RYAN Won’t Fund Border Fence For US Citiz...   
21215  WALMART MAKES SENIOR VETERAN Greeter Remove Mi...   
21216  BARACK OBAMA Finds Friend In “Fundamental Tran...   
21217  Baltimore Braces For Chaos: Mistrial Declared ...   
21218  GERMAN COURT RULES “Sharia Police” Patrolling ...   
...                                                  ...   
23476  McPain: John McCain Furious That Iran Treated ...   
23477  JUSTICE? Yahoo Settles E-mail Privacy Class-ac...   
23478  Sunnistan: US and Allied ‘Safe Zone’ Plan to T...   
23479  How to Blow $700 Million: Al Jazeera America F...   
23480  10 U.S. Navy Sailors Held by Iranian Military ...   

                                                    text      subject  \
21214  Ryan has never made any secret about his desir...    left-news   
21215  Lee Young, owner of Hitman Firearms posted thi...    left-news   
21216  Funding for Sanctuary Cities, benefits for ill...    left-news   
21217  The jurors in the Freddie Gray case were deadl...    left-news   
21218  These videos are very disturbing. Americans ar...    left-news   
...                                                  ...          ...   
23476  21st Century Wire says As 21WIRE reported earl...  Middle-east   
23477  21st Century Wire says It s a familiar theme. ...  Middle-east   
23478  Patrick Henningsen  21st Century WireRemember ...  Middle-east   
23479  21st Century Wire says Al Jazeera America will...  Middle-east   
23480  21st Century Wire says As 21WIRE predicted in ...  Middle-east   

                   date  y  
21214      Dec 17, 2015  0  
21215      Dec 17, 2015  0  
21216      Dec 17, 2015  0  
21217      Dec 16, 2015  0  
21218      Dec 16, 2015  0  
...                 ... ..  
23476  January 16, 2016  0  
23477  January 16, 2016  0  
23478  January 15, 2016  0  
23479  January 14, 2016  0  
23480  January 12, 2016  0  

[2267 rows x 5 columns]

In [19]:
# marge the two dataframs in one 
df = pd.concat([dft, dff1], ignore_index=True)

In [20]:
df

title  \
0      As U.S. budget fight looms, Republicans flip t...   
1      U.S. military to accept transgender recruits o...   
2      Senior U.S. Republican senator: 'Let Mr. Muell...   
3      FBI Russia probe helped by Australian diplomat...   
4      Trump wants Postal Service to charge 'much mor...   
...                                                  ...   
42417  ONLY IN DETROIT: Entitled Squatter Gets Squatt...   
42418  FOX NEWS REPORTER ASKS MAYOR Why She’s Using T...   
42419  NEW ACCUSATION AGAINST BERNIE SANDERS Exposes ...   
42420  JESUS CHRIST IS STRIPPED From Christmas Celebr...   
42421  Michael Moore Wants America To Know “We Are Al...   

                                                    text       subject  \
0      WASHINGTON (Reuters) - The head of a conservat...  politicsNews   
1      WASHINGTON (Reuters) - Transgender people will...  politicsNews   
2      WASHINGTON (Reuters) - The special counsel inv...  politicsNews   
3      WASHINGTON (Reuters) - Trump campaign adviser ...  politicsNews   
4      SEATTLE/WASHINGTON (Reuters) - President Donal...  politicsNews   
...                                                  ...           ...   
42417                   When karma bites ya  in the a*s      left-news   
42418  The drip drip drip of the Left ensuring Christ...     left-news   
42419  By now, it s pretty clear that Bernie doesn t ...     left-news   
42420  Does that mean Islam is not considered a faith...     left-news   
42421  No dumb ass  we are actually NOT all Muslims! ...     left-news   

                     date  y  
0      December 31, 2017   1  
1      December 29, 2017   1  
2      December 31, 2017   1  
3      December 30, 2017   1  
4      December 29, 2017   1  
...                   ... ..  
42417        Dec 18, 2015  0  
42418        Dec 18, 2015  0  
42419        Dec 18, 2015  0  
42420        Dec 18, 2015  0  
42421        Dec 17, 2015  0  

[42422 rows x 5 columns]

In [21]:
df = df.sample(frac=1, random_state=42).reset_index(drop=True)

In [22]:
df

title  \
0      BEYONCE DOUBLES DOWN…Debuts #LEMONADE, Another...   
1      McCain illness deprives Senate of crucial vote...   
2       REPORT: Civil Rights Groups Say Voter Suppres...   
3      DISNEY WORKER TELLS HORROR STORY OF BEING FORC...   
4      Trump 'weighed in' on son's Russia attorney st...   
...                                                  ...   
42417  Germany's Schaeuble warns Trump over protectio...   
42418  Twenty-eight arrests after Venezuela looting, ...   
42419  HOW THE FBI Cracked A Terror Plot On Black Fri...   
42420  U.S. tax plan threatens Wall Street's Bermuda ...   
42421  Asylum seekers refuse to leave Papua New Guine...   

                                                    text          subject  \
0      Most of the world will be obsessed with Beyonc...        left-news   
1      WASHINGTON (Reuters) - If John McCain’s illnes...     politicsNews   
2      In the wake of election day and Donald Trump s...             News   
3      We ve been screaming from the rooftop that Oba...         politics   
4      WASHINGTON (Reuters) - The White House said on...     politicsNews   
...                                                  ...              ...   
42417  BERLIN (Reuters) - German Finance Minister Wol...     politicsNews   
42418  PUERTO ORDAZ, Venezuela (Reuters) - Venezuelan...        worldnews   
42419   The U.S. is the great enemy of mankind!  rave...  Government News   
42420  NEW YORK (Reuters) - U.S. money managers with ...     politicsNews   
42421  SYDNEY (Reuters) - Hundreds of asylum seekers ...        worldnews   

                     date  y  
0            Apr 24, 2016  0  
1          July 20, 2017   1  
2       November 12, 2016  0  
3            May 21, 2015  0  
4         August 1, 2017   1  
...                   ... ..  
42417   January 16, 2017   1  
42418  December 26, 2017   1  
42419        Nov 28, 2015  0  
42420   November 2, 2017   1  
42421   October 31, 2017   1  

[42422 rows x 5 columns]

### the process of add the fueture of teh class and marge the two datasets is done (**is df**)
## the next stap is Reduction the fuetres

In [23]:
df = df.drop(columns=["subject", "date"])
df

title  \
0      BEYONCE DOUBLES DOWN…Debuts #LEMONADE, Another...   
1      McCain illness deprives Senate of crucial vote...   
2       REPORT: Civil Rights Groups Say Voter Suppres...   
3      DISNEY WORKER TELLS HORROR STORY OF BEING FORC...   
4      Trump 'weighed in' on son's Russia attorney st...   
...                                                  ...   
42417  Germany's Schaeuble warns Trump over protectio...   
42418  Twenty-eight arrests after Venezuela looting, ...   
42419  HOW THE FBI Cracked A Terror Plot On Black Fri...   
42420  U.S. tax plan threatens Wall Street's Bermuda ...   
42421  Asylum seekers refuse to leave Papua New Guine...   

                                                    text  y  
0      Most of the world will be obsessed with Beyonc...  0  
1      WASHINGTON (Reuters) - If John McCain’s illnes...  1  
2      In the wake of election day and Donald Trump s...  0  
3      We ve been screaming from the rooftop that Oba...  0  
4      WASHINGTON (Reuters) - The White House said on...  1  
...                                                  ... ..  
42417  BERLIN (Reuters) - German Finance Minister Wol...  1  
42418  PUERTO ORDAZ, Venezuela (Reuters) - Venezuelan...  1  
42419   The U.S. is the great enemy of mankind!  rave...  0  
42420  NEW YORK (Reuters) - U.S. money managers with ...  1  
42421  SYDNEY (Reuters) - Hundreds of asylum seekers ...  1  

[42422 rows x 3 columns]

In [24]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42422 entries, 0 to 42421
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   title   42422 non-null  object
 1   text    42422 non-null  object
 2   y       42422 non-null  int64 
dtypes: int64(1), object(2)
memory usage: 994.4+ KB
None


In [25]:
df["new"] = df["title"] + " " + df["text"]
df

title  \
0      BEYONCE DOUBLES DOWN…Debuts #LEMONADE, Another...   
1      McCain illness deprives Senate of crucial vote...   
2       REPORT: Civil Rights Groups Say Voter Suppres...   
3      DISNEY WORKER TELLS HORROR STORY OF BEING FORC...   
4      Trump 'weighed in' on son's Russia attorney st...   
...                                                  ...   
42417  Germany's Schaeuble warns Trump over protectio...   
42418  Twenty-eight arrests after Venezuela looting, ...   
42419  HOW THE FBI Cracked A Terror Plot On Black Fri...   
42420  U.S. tax plan threatens Wall Street's Bermuda ...   
42421  Asylum seekers refuse to leave Papua New Guine...   

                                                    text  y  \
0      Most of the world will be obsessed with Beyonc...  0   
1      WASHINGTON (Reuters) - If John McCain’s illnes...  1   
2      In the wake of election day and Donald Trump s...  0   
3      We ve been screaming from the rooftop that Oba...  0   
4      WASHINGTON (Reuters) - The White House said on...  1   
...                                                  ... ..   
42417  BERLIN (Reuters) - German Finance Minister Wol...  1   
42418  PUERTO ORDAZ, Venezuela (Reuters) - Venezuelan...  1   
42419   The U.S. is the great enemy of mankind!  rave...  0   
42420  NEW YORK (Reuters) - U.S. money managers with ...  1   
42421  SYDNEY (Reuters) - Hundreds of asylum seekers ...  1   

                                                     new  
0      BEYONCE DOUBLES DOWN…Debuts #LEMONADE, Another...  
1      McCain illness deprives Senate of crucial vote...  
2       REPORT: Civil Rights Groups Say Voter Suppres...  
3      DISNEY WORKER TELLS HORROR STORY OF BEING FORC...  
4      Trump 'weighed in' on son's Russia attorney st...  
...                                                  ...  
42417  Germany's Schaeuble warns Trump over protectio...  
42418  Twenty-eight arrests after Venezuela looting, ...  
42419  HOW THE FBI Cracked A Terror Plot On Black Fri...  
42420  U.S. tax plan threatens Wall Street's Bermuda ...  
42421  Asylum seekers refuse to leave Papua New Guine...  

[42422 rows x 4 columns]

In [26]:
df = df.drop(columns=["title", "text"])
df

y                                                new
0      0  BEYONCE DOUBLES DOWN…Debuts #LEMONADE, Another...
1      1  McCain illness deprives Senate of crucial vote...
2      0   REPORT: Civil Rights Groups Say Voter Suppres...
3      0  DISNEY WORKER TELLS HORROR STORY OF BEING FORC...
4      1  Trump 'weighed in' on son's Russia attorney st...
...   ..                                                ...
42417  1  Germany's Schaeuble warns Trump over protectio...
42418  1  Twenty-eight arrests after Venezuela looting, ...
42419  0  HOW THE FBI Cracked A Terror Plot On Black Fri...
42420  1  U.S. tax plan threatens Wall Street's Bermuda ...
42421  1  Asylum seekers refuse to leave Papua New Guine...

[42422 rows x 2 columns]

## now we finished the stap of the Reduction the rows 
### now the need to normalisate the data 

In [27]:
df["text"] = (
    df["new"]
    # 1. حذف الروابط URLs
    .str.replace(r'http\S+|www\.\S+', ' ', regex=True)

    # 2. حذف الأرقام
    .str.replace(r'\d+', ' ', regex=True)

    # 3. استبدال الرموز غير الحروف والمسافات بمسافة
    .str.replace(r'[^\w\s]', ' ', regex=True)

    # 4. إزالة المسافات المكررة
    .str.replace(r'\s+', ' ', regex=True)

    # 5. إزالة المسافة في بداية/نهاية النص
    .str.strip()
)


In [28]:
df

y                                                new  \
0      0  BEYONCE DOUBLES DOWN…Debuts #LEMONADE, Another...   
1      1  McCain illness deprives Senate of crucial vote...   
2      0   REPORT: Civil Rights Groups Say Voter Suppres...   
3      0  DISNEY WORKER TELLS HORROR STORY OF BEING FORC...   
4      1  Trump 'weighed in' on son's Russia attorney st...   
...   ..                                                ...   
42417  1  Germany's Schaeuble warns Trump over protectio...   
42418  1  Twenty-eight arrests after Venezuela looting, ...   
42419  0  HOW THE FBI Cracked A Terror Plot On Black Fri...   
42420  1  U.S. tax plan threatens Wall Street's Bermuda ...   
42421  1  Asylum seekers refuse to leave Papua New Guine...   

                                                    text  
0      BEYONCE DOUBLES DOWN Debuts LEMONADE Another R...  
1      McCain illness deprives Senate of crucial vote...  
2      REPORT Civil Rights Groups Say Voter Suppressi...  
3      DISNEY WORKER TELLS HORROR STORY OF BEING FORC...  
4      Trump weighed in on son s Russia attorney stat...  
...                                                  ...  
42417  Germany s Schaeuble warns Trump over protectio...  
42418  Twenty eight arrests after Venezuela looting v...  
42419  HOW THE FBI Cracked A Terror Plot On Black Fri...  
42420  U S tax plan threatens Wall Street s Bermuda i...  
42421  Asylum seekers refuse to leave Papua New Guine...  

[42422 rows x 3 columns]

In [29]:
from nltk.corpus import stopwords
from nltk.tokenize import word_tokenize
import nltk

In [30]:
nltk.download('stopwords')
nltk.download('punkt')
nltk.download("punkt_tab")

[nltk_data] Downloading package stopwords to
[nltk_data]     C:\Users\DELL\AppData\Roaming\nltk_data...
[nltk_data]   Package stopwords is already up-to-date!
[nltk_data] Downloading package punkt to
[nltk_data]     C:\Users\DELL\AppData\Roaming\nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package punkt_tab to
[nltk_data]     C:\Users\DELL\AppData\Roaming\nltk_data...
[nltk_data]   Package punkt_tab is already up-to-date!


True

In [31]:
stop_words = set(stopwords.words('english'))

In [32]:
def remove_stop_words(sentence):
    words = word_tokenize(sentence)
    return " ".join([word for word in words if word.lower() not in stop_words])

In [33]:
df["clean_text"] = df["text"].apply(remove_stop_words)
df

y                                                new  \
0      0  BEYONCE DOUBLES DOWN…Debuts #LEMONADE, Another...   
1      1  McCain illness deprives Senate of crucial vote...   
2      0   REPORT: Civil Rights Groups Say Voter Suppres...   
3      0  DISNEY WORKER TELLS HORROR STORY OF BEING FORC...   
4      1  Trump 'weighed in' on son's Russia attorney st...   
...   ..                                                ...   
42417  1  Germany's Schaeuble warns Trump over protectio...   
42418  1  Twenty-eight arrests after Venezuela looting, ...   
42419  0  HOW THE FBI Cracked A Terror Plot On Black Fri...   
42420  1  U.S. tax plan threatens Wall Street's Bermuda ...   
42421  1  Asylum seekers refuse to leave Papua New Guine...   

                                                    text  \
0      BEYONCE DOUBLES DOWN Debuts LEMONADE Another R...   
1      McCain illness deprives Senate of crucial vote...   
2      REPORT Civil Rights Groups Say Voter Suppressi...   
3      DISNEY WORKER TELLS HORROR STORY OF BEING FORC...   
4      Trump weighed in on son s Russia attorney stat...   
...                                                  ...   
42417  Germany s Schaeuble warns Trump over protectio...   
42418  Twenty eight arrests after Venezuela looting v...   
42419  HOW THE FBI Cracked A Terror Plot On Black Fri...   
42420  U S tax plan threatens Wall Street s Bermuda i...   
42421  Asylum seekers refuse to leave Papua New Guine...   

                                              clean_text  
0      BEYONCE DOUBLES Debuts LEMONADE Another Race B...  
1      McCain illness deprives Senate crucial vote Tr...  
2      REPORT Civil Rights Groups Say Voter Suppressi...  
3      DISNEY WORKER TELLS HORROR STORY FORCED TRAIN ...  
4      Trump weighed son Russia attorney statement Wh...  
...                                                  ...  
42417  Germany Schaeuble warns Trump protectionism WS...  
42418  Twenty eight arrests Venezuela looting violenc...  
42419  FBI Cracked Terror Plot Black Friday May Worse...  
42420  U tax plan threatens Wall Street Bermuda idyll...  
42421  Asylum seekers refuse leave Papua New Guinea c...  

[42422 rows x 4 columns]

In [34]:
def remove_single_char_words(sentence):
    # نحتفظ بالكلمات التي طولها أكثر من حرف واحد
    return " ".join([word for word in sentence.split() if len(word) > 1])

df["clean_text"] = df["clean_text"].apply(remove_single_char_words)

In [35]:
df

y                                                new  \
0      0  BEYONCE DOUBLES DOWN…Debuts #LEMONADE, Another...   
1      1  McCain illness deprives Senate of crucial vote...   
2      0   REPORT: Civil Rights Groups Say Voter Suppres...   
3      0  DISNEY WORKER TELLS HORROR STORY OF BEING FORC...   
4      1  Trump 'weighed in' on son's Russia attorney st...   
...   ..                                                ...   
42417  1  Germany's Schaeuble warns Trump over protectio...   
42418  1  Twenty-eight arrests after Venezuela looting, ...   
42419  0  HOW THE FBI Cracked A Terror Plot On Black Fri...   
42420  1  U.S. tax plan threatens Wall Street's Bermuda ...   
42421  1  Asylum seekers refuse to leave Papua New Guine...   

                                                    text  \
0      BEYONCE DOUBLES DOWN Debuts LEMONADE Another R...   
1      McCain illness deprives Senate of crucial vote...   
2      REPORT Civil Rights Groups Say Voter Suppressi...   
3      DISNEY WORKER TELLS HORROR STORY OF BEING FORC...   
4      Trump weighed in on son s Russia attorney stat...   
...                                                  ...   
42417  Germany s Schaeuble warns Trump over protectio...   
42418  Twenty eight arrests after Venezuela looting v...   
42419  HOW THE FBI Cracked A Terror Plot On Black Fri...   
42420  U S tax plan threatens Wall Street s Bermuda i...   
42421  Asylum seekers refuse to leave Papua New Guine...   

                                              clean_text  
0      BEYONCE DOUBLES Debuts LEMONADE Another Race B...  
1      McCain illness deprives Senate crucial vote Tr...  
2      REPORT Civil Rights Groups Say Voter Suppressi...  
3      DISNEY WORKER TELLS HORROR STORY FORCED TRAIN ...  
4      Trump weighed son Russia attorney statement Wh...  
...                                                  ...  
42417  Germany Schaeuble warns Trump protectionism WS...  
42418  Twenty eight arrests Venezuela looting violenc...  
42419  FBI Cracked Terror Plot Black Friday May Worse...  
42420  tax plan threatens Wall Street Bermuda idyll N...  
42421  Asylum seekers refuse leave Papua New Guinea c...  

[42422 rows x 4 columns]

In [36]:
df["clean_text"] = df["clean_text"].str.lower()

In [37]:
df

y                                                new  \
0      0  BEYONCE DOUBLES DOWN…Debuts #LEMONADE, Another...   
1      1  McCain illness deprives Senate of crucial vote...   
2      0   REPORT: Civil Rights Groups Say Voter Suppres...   
3      0  DISNEY WORKER TELLS HORROR STORY OF BEING FORC...   
4      1  Trump 'weighed in' on son's Russia attorney st...   
...   ..                                                ...   
42417  1  Germany's Schaeuble warns Trump over protectio...   
42418  1  Twenty-eight arrests after Venezuela looting, ...   
42419  0  HOW THE FBI Cracked A Terror Plot On Black Fri...   
42420  1  U.S. tax plan threatens Wall Street's Bermuda ...   
42421  1  Asylum seekers refuse to leave Papua New Guine...   

                                                    text  \
0      BEYONCE DOUBLES DOWN Debuts LEMONADE Another R...   
1      McCain illness deprives Senate of crucial vote...   
2      REPORT Civil Rights Groups Say Voter Suppressi...   
3      DISNEY WORKER TELLS HORROR STORY OF BEING FORC...   
4      Trump weighed in on son s Russia attorney stat...   
...                                                  ...   
42417  Germany s Schaeuble warns Trump over protectio...   
42418  Twenty eight arrests after Venezuela looting v...   
42419  HOW THE FBI Cracked A Terror Plot On Black Fri...   
42420  U S tax plan threatens Wall Street s Bermuda i...   
42421  Asylum seekers refuse to leave Papua New Guine...   

                                              clean_text  
0      beyonce doubles debuts lemonade another race b...  
1      mccain illness deprives senate crucial vote tr...  
2      report civil rights groups say voter suppressi...  
3      disney worker tells horror story forced train ...  
4      trump weighed son russia attorney statement wh...  
...                                                  ...  
42417  germany schaeuble warns trump protectionism ws...  
42418  twenty eight arrests venezuela looting violenc...  
42419  fbi cracked terror plot black friday may worse...  
42420  tax plan threatens wall street bermuda idyll n...  
42421  asylum seekers refuse leave papua new guinea c...  

[42422 rows x 4 columns]

In [38]:
df = df.drop(columns=["new", "text"])

In [39]:
df

y                                         clean_text
0      0  beyonce doubles debuts lemonade another race b...
1      1  mccain illness deprives senate crucial vote tr...
2      0  report civil rights groups say voter suppressi...
3      0  disney worker tells horror story forced train ...
4      1  trump weighed son russia attorney statement wh...
...   ..                                                ...
42417  1  germany schaeuble warns trump protectionism ws...
42418  1  twenty eight arrests venezuela looting violenc...
42419  0  fbi cracked terror plot black friday may worse...
42420  1  tax plan threatens wall street bermuda idyll n...
42421  1  asylum seekers refuse leave papua new guinea c...

[42422 rows x 2 columns]

In [40]:
# dfc = df
# from deep_translator import GoogleTranslator
# from tqdm.auto import tqdm
# tqdm.pandas()
# def translate_long_text(text, max_chars=500):
#     translated_text = ""
#     for i in range(0, len(text), max_chars):
#         part = text[i:i+max_chars]
#         translated_text += GoogleTranslator(source='en', target='ar').translate(part)
#     return translated_text
# def translate_text(text):
#     try:
#         return translator.translate(text, src='en', dest='ar').text
#     except:
#         return text

# dfc['text_ar'] = dfc['clean_text'].progress_apply(translate_text)


  0%|          | 0/42422 [00:00<?, ?it/s]

In [46]:
import asyncio
from googletrans import Translator
from tqdm.auto import tqdm

translator = Translator()

async def translate_text_async(text):
    try:
        result = await translator.translate(text, src='en', dest='ar')
        return result.text
    except:
        return text

async def translate_column(df):
    result = []
    for text in tqdm(df['clean_text'], total=len(df)):
        translated = await translate_text_async(text)
        result.append(translated)
    df['text_ar'] = result
    return df

import nest_asyncio
nest_asyncio.apply()

dfc = asyncio.get_event_loop().run_until_complete(translate_column(dfc))


  0%|          | 0/42422 [00:00<?, ?it/s]

KeyboardInterrupt: 

In [47]:
dfc 

y                                         clean_text  \
0      0  beyonce doubles debuts lemonade another race b...   
1      1  mccain illness deprives senate crucial vote tr...   
2      0  report civil rights groups say voter suppressi...   
3      0  disney worker tells horror story forced train ...   
4      1  trump weighed son russia attorney statement wh...   
...   ..                                                ...   
42417  1  germany schaeuble warns trump protectionism ws...   
42418  1  twenty eight arrests venezuela looting violenc...   
42419  0  fbi cracked terror plot black friday may worse...   
42420  1  tax plan threatens wall street bermuda idyll n...   
42421  1  asylum seekers refuse leave papua new guinea c...   

                                                 text_ar  
0      beyonce doubles debuts lemonade another race b...  
1      mccain illness deprives senate crucial vote tr...  
2      report civil rights groups say voter suppressi...  
3      disney worker tells horror story forced train ...  
4      trump weighed son russia attorney statement wh...  
...                                                  ...  
42417  germany schaeuble warns trump protectionism ws...  
42418  twenty eight arrests venezuela looting violenc...  
42419  fbi cracked terror plot black friday may worse...  
42420  tax plan threatens wall street bermuda idyll n...  
42421  asylum seekers refuse leave papua new guinea c...  

[42422 rows x 3 columns]

In [49]:
dftt = pd.read_csv('translated_texts.csv')
dftt

y                                         clean_text  \
0      0  beyonce doubles debuts lemonade another race b...   
1      1  mccain illness deprives senate crucial vote tr...   
2      0  report civil rights groups say voter suppressi...   
3      0  disney worker tells horror story forced train ...   
4      1  trump weighed son russia attorney statement wh...   
...   ..                                                ...   
42417  1  germany schaeuble warns trump protectionism ws...   
42418  1  twenty eight arrests venezuela looting violenc...   
42419  0  fbi cracked terror plot black friday may worse...   
42420  1  tax plan threatens wall street bermuda idyll n...   
42421  1  asylum seekers refuse leave papua new guinea c...   

                                                 text_ar  
0      بيونسيه الزوجي لاول مرة عصير الليمون سباق آخر ...  
1      مرض ماكين يحرم مجلس الشيوخ من التصويت الحاسم ل...  
2      تقرير جماعات الحقوق المدنية يقول إن قمع الناخب...  
3      يروي عامل ديزني قصة رعب، قطار قسري، عامل أجنبي...  
4      ترامب يزن ابنه روسيا بيان المدعي العام البيت ا...  
...                                                  ...  
42417  ألمانيا يحذر شيوبله من حمائية ترامب، وول ستريت...  
42418  28 اعتقالات في فنزويلا، نهب، عنف، بورتو أورداز...  
42419  كشف مكتب التحقيقات الفيدرالي مؤامرة إرهابية، ا...  
42420  خطة الضرائب تهدد وول ستريت برمودا الشاعرة نيوي...  
42421  يرفض طالبو اللجوء مغادرة مخيم بابوا غينيا الجد...  

[42422 rows x 3 columns]

In [50]:
print(dftt["text_ar"][0])

بيونسيه الزوجي لاول مرة عصير الليمون سباق آخر اصطياد شرطي يكره أغنية مؤيدة أوباما حلوة مايكل براون ماما سيرينا ويليامز العالم مهووس بيونسيه إصدار جديد عصير الليمون اعتراف غير متوقع متزوج غش خنزير بدلاً من ذلك تم اختيار التركيز بدلاً من ذلك مواصلة الجهود الديناميكية الثنائي العنصري تعزيز عدم الثقة الكراهية رجال إنفاذ القانون مساعدة الصديق العزيز باراك أوباما تعزيز الانقسام العنصري أمريكا بيونسيه تصوير الذات قديس قيادة النساء المعمودية مضحك بالنظر إلى رسائل الكراهية المثيرة للانقسام تستمر في الترويج للأغاني تصبح لا شيء أداة أوباما حياة السود مهمة مجموعة إرهابية أي شخص يشتري موسيقى زوج علامة عنصرية يقدم مساهمة مباشرة أكره الأمهات ترايفون مارتن مايكل براون ظهرت بيونسيه ألبوم جديد تم إصداره جاي تيدال خدمة البث ليلة السبت HBO خاص ترويج عصير الليمون عنوان المغني الألبوم الأول ثلاث سنوات ألبوم مرئي تم بثه HBO تعزيز المغني استمرار الدانتيل العمل رسالة سياسية ضرب الزوج جاي وسط شائعات الخيانة الزوجية الظهور في النهاية سامح الفيلم بثت HBO خدم جريئة تحية مثابرة امرأة أمريكية من أصل أفريقي موضوع حيا

In [52]:
print(dftt["text_ar"][5])
print(dftt["clean_text"][5])

تقدم كلينتون على ترامب يضيق أقل من ثلاث نقاط استطلاع رويترز إيبسوس نيويورك رويترز المرشح الرئاسي الديمقراطي هيلاري كلينتون تتقدم على منافسها الجمهوري دونالد ترامب ضاقت نقاط مئوية أقل وفقًا لاستطلاع الرأي الذي أجرته رويترز إيبسوس يوم الجمعة ما يقرب من ثماني نقاط في المائة يوم الاثنين في المائة من المحتمل أن يفضل الناخبون كلينتون ترامب في المائة وفقًا لاستطلاع يوليو/تموز وأغسطس على الإنترنت، فاصل مصداقية استطلاع الناخبين المحتملين زائد ناقص نقاط مئوية مما يعني أن النتائج تشير إلى أن السباق متساوي تقريبًا بين الناخبين المسجلين في الفترة التي احتلت فيها كلينتون الصدارة بنسبة خمسة في المائة نقاط ثماني نقاط مئوية، الاثنين، وفقًا لاستطلاع الرأي، أسباب التحول غير واضحة، كلينتون تقدمت بفارق كبير على ترامب، في أعقاب المؤتمر الوطني الديمقراطي، الأسبوع الماضي، أصبحت أول امرأة تقبل الترشيح الرئاسي، حزب سياسي كبير منذ أيام، عداء طويل، عائلة، جندي أمريكي، قتل في العراق، تشاجرت القيادة الجمهورية، تعليقات القيادة الجمهورية، اضطراب القيادة داخل حملة ترامب في الأيام الأخيرة، ومع ذلك سعى إلى إعادة التركيز

In [56]:
import re
def remove_english_words(text):
    # حذف الكلمات الإنجليزية
    text = re.sub(r'\b[a-zA-Z]+\b', '', text)
    # إزالة الفراغات الزائدة
    text = re.sub(r'\s+', ' ', text).strip()
    return text

# تطبيق الدالة على العمود
dftt['clean_text'] = dftt['text_ar'].apply(remove_english_words)

TypeError: expected string or bytes-like object, got 'float'

In [57]:
import re
import pandas as pd

def remove_english_words(text):
    if not isinstance(text, str):  # التأكد أن النص فعلاً string
        return ''
    # حذف الكلمات الإنجليزية
    text = re.sub(r'\b[a-zA-Z]+\b', '', text)
    # إزالة الفراغات الزائدة
    text = re.sub(r'\s+', ' ', text).strip()
    return text

# تطبيق الدالة على العمود
dftt['clean_text'] = dftt['text_ar'].apply(remove_english_words)


In [58]:
dftt

y                                         clean_text  \
0      0  بيونسيه الزوجي لاول مرة عصير الليمون سباق آخر ...   
1      1  مرض ماكين يحرم مجلس الشيوخ من التصويت الحاسم ل...   
2      0  تقرير جماعات الحقوق المدنية يقول إن قمع الناخب...   
3      0  يروي عامل ديزني قصة رعب، قطار قسري، عامل أجنبي...   
4      1  ترامب يزن ابنه روسيا بيان المدعي العام البيت ا...   
...   ..                                                ...   
42417  1  ألمانيا يحذر شيوبله من حمائية ترامب، وول ستريت...   
42418  1  28 اعتقالات في فنزويلا، نهب، عنف، بورتو أورداز...   
42419  0  كشف مكتب التحقيقات الفيدرالي مؤامرة إرهابية، ا...   
42420  1  خطة الضرائب تهدد وول ستريت برمودا الشاعرة نيوي...   
42421  1  يرفض طالبو اللجوء مغادرة مخيم بابوا غينيا الجد...   

                                                 text_ar  
0      بيونسيه الزوجي لاول مرة عصير الليمون سباق آخر ...  
1      مرض ماكين يحرم مجلس الشيوخ من التصويت الحاسم ل...  
2      تقرير جماعات الحقوق المدنية يقول إن قمع الناخب...  
3      يروي عامل ديزني قصة رعب، قطار قسري، عامل أجنبي...  
4      ترامب يزن ابنه روسيا بيان المدعي العام البيت ا...  
...                                                  ...  
42417  ألمانيا يحذر شيوبله من حمائية ترامب، وول ستريت...  
42418  28 اعتقالات في فنزويلا، نهب، عنف، بورتو أورداز...  
42419  كشف مكتب التحقيقات الفيدرالي مؤامرة إرهابية، ا...  
42420  خطة الضرائب تهدد وول ستريت برمودا الشاعرة نيوي...  
42421  يرفض طالبو اللجوء مغادرة مخيم بابوا غينيا الجد...  

[42422 rows x 3 columns]

In [59]:
dftt["clean_text"][3]

'يروي عامل ديزني قصة رعب، قطار قسري، عامل أجنبي، يحل محل حزمة تعويضات إنهاء الخدمة، يصرخ على السطح، أوباما، سياسة الحدود المفتوحة الجديدة، تدمر فرص العمل، الأمريكيون يريدون حقًا العمل، يسمعوننا نحافظ على ديزني المشار إليها في أسعد مكان على وجه الأرض لفترة أطول بكثير، السناتور الجمهوري، المرشح الرئاسي، ماركو روبيو، دعم مشروع قانون من شأنه أن يضاعف عدد العمال الضيوف، يمكن للشركات توظيفهم كل عام، مئات العمال الذين تم طردهم من الدولة حرفيًا، استبدلوا العمال الضيوف الأجانب، ديزني جنوب كاليفورنيا، أديسون مؤخرًا مجموعة الأحفوري معًا، طردت مئات عمال التكنولوجيا الأمريكيين، أجبروا على تدريب البدلاء الأجانب، العديد منهم يسافرون على وجه التحديد للحصول على وظيفة في يوم من الأيام في اليوم التالي الهندي عامل ذو مستوى مهارة أقل يجلس على مكتب، مئات من عمال التكنولوجيا الذين تم فصلهم مؤخرًا من شركة ديزني، أخبروا مؤسسة أخبار المتصل اليومية، أبلغ مئات زملائهم من أعضاء فريق التمثيل في أكتوبر الماضي، يمكن لقوة العمل الأجنبية المستبدلة إما البقاء لمدة أيام، استبدال القطار، حسن الخلق، المغادرة على الفور، وال

In [60]:
dftt["text_ar"][3]

'يروي عامل ديزني قصة رعب، قطار قسري، عامل أجنبي، يحل محل حزمة تعويضات إنهاء الخدمة، يصرخ على السطح، أوباما، سياسة الحدود المفتوحة الجديدة، تدمر فرص العمل، الأمريكيون يريدون حقًا العمل، يسمعوننا نحافظ على ديزني المشار إليها في أسعد مكان على وجه الأرض لفترة أطول بكثير، السناتور الجمهوري، المرشح الرئاسي، ماركو روبيو، دعم مشروع قانون من شأنه أن يضاعف عدد العمال الضيوف، يمكن للشركات توظيفهم كل عام، مئات العمال الذين تم طردهم من الدولة حرفيًا، استبدلوا العمال الضيوف الأجانب، ديزني جنوب كاليفورنيا، أديسون مؤخرًا مجموعة الأحفوري معًا، طردت مئات عمال التكنولوجيا الأمريكيين، أجبروا على تدريب البدلاء الأجانب، العديد منهم يسافرون على وجه التحديد للحصول على وظيفة في يوم من الأيام في اليوم التالي الهندي عامل ذو مستوى مهارة أقل يجلس على مكتب، مئات من عمال التكنولوجيا الذين تم فصلهم مؤخرًا من شركة ديزني، أخبروا مؤسسة أخبار المتصل اليومية، أبلغ مئات زملائهم من أعضاء فريق التمثيل في أكتوبر الماضي، يمكن لقوة العمل الأجنبية المستبدلة إما البقاء لمدة أيام، استبدال القطار، حسن الخلق، المغادرة على الفور، وال

In [64]:
dftt["clean_text"][10]

'يجب أن تشاهد باراك أوباما، يوم الـ 10 من كانون الثاني (يناير)، يأتي في وقت قريب بما فيه الكفاية، بحاجة إلى متعاطف مع الإرهاب الأمريكي، في البيت الأبيض، أسوأ رئيس على الإطلاق، ، ستيف هيرش، ستيفنويرش، ديسمبر'

# the clening data in the column (clean_text)

In [71]:
fd = pd.read_csv('translated_texts (1).csv')

In [72]:
fd

y                                               text  \
0      0  beyonce doubles down…debuts #lemonade, another...   
1      1  mccain illness deprives senate of crucial vote...   
2      0  report: civil rights groups say voter suppress...   
3      0  disney worker tells horror story of being forc...   
4      1  trump 'weighed in' on son's russia attorney st...   
...   ..                                                ...   
42417  1  germany's schaeuble warns trump over protectio...   
42418  1  twenty-eight arrests after venezuela looting, ...   
42419  0  how the fbi cracked a terror plot on black fri...   
42420  1  u.s. tax plan threatens wall street's bermuda ...   
42421  1  asylum seekers refuse to leave papua new guine...   

                                                 text_ar  
0      بيونسيه تتضاعف... لأول مرة #lemonade، أغنية أخ...  
1      مرض ماكين يحرم مجلس الشيوخ من التصويت الحاسم، ...  
2      تقرير: تقول جماعات الحقوق المدنية إن قمع الناخ...  
3      يروي عامل ديزني قصة مرعبة عن إجباره على تدريب ...  
4      البيت الأبيض: واشنطن (رويترز) - قال البيت الأب...  
...                                                  ...  
42417  وزير المالية الألماني فولفغانغ شيوبله يحذر ترا...  
42418  بورتو أورداز (فنزويلا) (رويترز) - قالت السلطات...  
42419  كيف تمكن مكتب التحقيقات الفيدرالي من كشف مؤامر...  
42420  نحن. خطة ضريبية تهدد منطقة برمودا في وول ستريت...  
42421  رفض مئات من طالبي اللجوء مغادرة مركز احتجاز في...  

[42422 rows x 3 columns]

In [73]:
fd['text_ar'][0]

'بيونسيه تتضاعف... لأول مرة #lemonade، أغنية أخرى لاستدراج العرق، كراهية الشرطيين، مؤيدة لأوباما... مع والدة مايكل براون الجميلة وسيرينا ويليامز، سيكون معظم العالم مهووسًا بإصدار بيونسيه الجديد، عصير الليمون بسبب اعترافها غير المتوقع بأنها متزوجة من خنزير غشاش. لقد اخترنا بدلاً من ذلك التركيز على الجهود المستمرة التي يبذلها الثنائي العنصري الديناميكي لتعزيز عدم الثقة والكراهية تجاه رجال إنفاذ القانون لدينا، بينما نساعد صديقهم العزيز باراك أوباما على تعزيز الانقسام العنصري في أمريكا. إن تصوير بيونسيه لنفسها كقديسة تقود النساء إلى المعمودية أمر مثير للضحك بالنظر إلى رسائل الكراهية والخلاف التي تواصل الترويج لها في أغانيها. لقد أصبحت مجرد أداة لأوباما وحياة السود مهمة للجماعة الإرهابية. أي شخص يشتري موسيقاها من شركة زوجها العنصرية فهو يساهم بشكل مباشر في الكراهية. ظهرت والدا ترايفون مارتن ومايكل براون في الألبوم الجديد لبيونسيه، والذي تم إصداره على خدمة جاي زي تيدال للبث المباشر مساء السبت بعد عرض خاص على قناة HBO للترويج له. عصير الليمون هو عنوان الألبوم الأول للمغنية منذ ثلاث سنوات، وال

In [77]:
fd = fd.drop(columns=["clean_text"])

In [78]:
fd

y                                               text  \
0      0  beyonce doubles down…debuts #lemonade, another...   
1      1  mccain illness deprives senate of crucial vote...   
2      0  report: civil rights groups say voter suppress...   
3      0  disney worker tells horror story of being forc...   
4      1  trump 'weighed in' on son's russia attorney st...   
...   ..                                                ...   
42417  1  germany's schaeuble warns trump over protectio...   
42418  1  twenty-eight arrests after venezuela looting, ...   
42419  0  how the fbi cracked a terror plot on black fri...   
42420  1  u.s. tax plan threatens wall street's bermuda ...   
42421  1  asylum seekers refuse to leave papua new guine...   

                                                 text_ar  
0      بيونسيه تتضاعف... لأول مرة #lemonade، أغنية أخ...  
1      مرض ماكين يحرم مجلس الشيوخ من التصويت الحاسم، ...  
2      تقرير: تقول جماعات الحقوق المدنية إن قمع الناخ...  
3      يروي عامل ديزني قصة مرعبة عن إجباره على تدريب ...  
4      البيت الأبيض: واشنطن (رويترز) - قال البيت الأب...  
...                                                  ...  
42417  وزير المالية الألماني فولفغانغ شيوبله يحذر ترا...  
42418  بورتو أورداز (فنزويلا) (رويترز) - قالت السلطات...  
42419  كيف تمكن مكتب التحقيقات الفيدرالي من كشف مؤامر...  
42420  نحن. خطة ضريبية تهدد منطقة برمودا في وول ستريت...  
42421  رفض مئات من طالبي اللجوء مغادرة مركز احتجاز في...  

[42422 rows x 3 columns]

In [81]:
def clean_arabic_text(text):
    
    if pd.isna(text) or text is None:
        return ""
    
    text = str(text)
    cleaned_text = re.sub(r'[^\u0600-\u06FF\s]+', ' ', text)
    
    cleaned_text = re.sub(r'\s+', ' ', cleaned_text).strip()
    
    return cleaned_text


fd['Cleaned_Text'] = fd['text_ar'].apply(clean_arabic_text)



In [82]:
fd

y                                               text  \
0      0  beyonce doubles down…debuts #lemonade, another...   
1      1  mccain illness deprives senate of crucial vote...   
2      0  report: civil rights groups say voter suppress...   
3      0  disney worker tells horror story of being forc...   
4      1  trump 'weighed in' on son's russia attorney st...   
...   ..                                                ...   
42417  1  germany's schaeuble warns trump over protectio...   
42418  1  twenty-eight arrests after venezuela looting, ...   
42419  0  how the fbi cracked a terror plot on black fri...   
42420  1  u.s. tax plan threatens wall street's bermuda ...   
42421  1  asylum seekers refuse to leave papua new guine...   

                                                 text_ar  \
0      بيونسيه تتضاعف... لأول مرة #lemonade، أغنية أخ...   
1      مرض ماكين يحرم مجلس الشيوخ من التصويت الحاسم، ...   
2      تقرير: تقول جماعات الحقوق المدنية إن قمع الناخ...   
3      يروي عامل ديزني قصة مرعبة عن إجباره على تدريب ...   
4      البيت الأبيض: واشنطن (رويترز) - قال البيت الأب...   
...                                                  ...   
42417  وزير المالية الألماني فولفغانغ شيوبله يحذر ترا...   
42418  بورتو أورداز (فنزويلا) (رويترز) - قالت السلطات...   
42419  كيف تمكن مكتب التحقيقات الفيدرالي من كشف مؤامر...   
42420  نحن. خطة ضريبية تهدد منطقة برمودا في وول ستريت...   
42421  رفض مئات من طالبي اللجوء مغادرة مركز احتجاز في...   

                                            Cleaned_Text  
0      بيونسيه تتضاعف لأول مرة ، أغنية أخرى لاستدراج ...  
1      مرض ماكين يحرم مجلس الشيوخ من التصويت الحاسم، ...  
2      تقرير تقول جماعات الحقوق المدنية إن قمع الناخب...  
3      يروي عامل ديزني قصة مرعبة عن إجباره على تدريب ...  
4      البيت الأبيض واشنطن رويترز قال البيت الأبيض يو...  
...                                                  ...  
42417  وزير المالية الألماني فولفغانغ شيوبله يحذر ترا...  
42418  بورتو أورداز فنزويلا رويترز قالت السلطات الفنز...  
42419  كيف تمكن مكتب التحقيقات الفيدرالي من كشف مؤامر...  
42420  نحن خطة ضريبية تهدد منطقة برمودا في وول ستريت ...  
42421  رفض مئات من طالبي اللجوء مغادرة مركز احتجاز في...  

[42422 rows x 4 columns]

In [83]:
fd['Cleaned_Text'] = fd['Cleaned_Text'].apply(clean_arabic_text)

In [84]:
fd

y                                               text  \
0      0  beyonce doubles down…debuts #lemonade, another...   
1      1  mccain illness deprives senate of crucial vote...   
2      0  report: civil rights groups say voter suppress...   
3      0  disney worker tells horror story of being forc...   
4      1  trump 'weighed in' on son's russia attorney st...   
...   ..                                                ...   
42417  1  germany's schaeuble warns trump over protectio...   
42418  1  twenty-eight arrests after venezuela looting, ...   
42419  0  how the fbi cracked a terror plot on black fri...   
42420  1  u.s. tax plan threatens wall street's bermuda ...   
42421  1  asylum seekers refuse to leave papua new guine...   

                                                 text_ar  \
0      بيونسيه تتضاعف... لأول مرة #lemonade، أغنية أخ...   
1      مرض ماكين يحرم مجلس الشيوخ من التصويت الحاسم، ...   
2      تقرير: تقول جماعات الحقوق المدنية إن قمع الناخ...   
3      يروي عامل ديزني قصة مرعبة عن إجباره على تدريب ...   
4      البيت الأبيض: واشنطن (رويترز) - قال البيت الأب...   
...                                                  ...   
42417  وزير المالية الألماني فولفغانغ شيوبله يحذر ترا...   
42418  بورتو أورداز (فنزويلا) (رويترز) - قالت السلطات...   
42419  كيف تمكن مكتب التحقيقات الفيدرالي من كشف مؤامر...   
42420  نحن. خطة ضريبية تهدد منطقة برمودا في وول ستريت...   
42421  رفض مئات من طالبي اللجوء مغادرة مركز احتجاز في...   

                                            Cleaned_Text  
0      بيونسيه تتضاعف لأول مرة ، أغنية أخرى لاستدراج ...  
1      مرض ماكين يحرم مجلس الشيوخ من التصويت الحاسم، ...  
2      تقرير تقول جماعات الحقوق المدنية إن قمع الناخب...  
3      يروي عامل ديزني قصة مرعبة عن إجباره على تدريب ...  
4      البيت الأبيض واشنطن رويترز قال البيت الأبيض يو...  
...                                                  ...  
42417  وزير المالية الألماني فولفغانغ شيوبله يحذر ترا...  
42418  بورتو أورداز فنزويلا رويترز قالت السلطات الفنز...  
42419  كيف تمكن مكتب التحقيقات الفيدرالي من كشف مؤامر...  
42420  نحن خطة ضريبية تهدد منطقة برمودا في وول ستريت ...  
42421  رفض مئات من طالبي اللجوء مغادرة مركز احتجاز في...  

[42422 rows x 4 columns]

In [85]:
fd['Cleaned_Text'][0]

'بيونسيه تتضاعف لأول مرة ، أغنية أخرى لاستدراج العرق، كراهية الشرطيين، مؤيدة لأوباما مع والدة مايكل براون الجميلة وسيرينا ويليامز، سيكون معظم العالم مهووسًا بإصدار بيونسيه الجديد، عصير الليمون بسبب اعترافها غير المتوقع بأنها متزوجة من خنزير غشاش لقد اخترنا بدلاً من ذلك التركيز على الجهود المستمرة التي يبذلها الثنائي العنصري الديناميكي لتعزيز عدم الثقة والكراهية تجاه رجال إنفاذ القانون لدينا، بينما نساعد صديقهم العزيز باراك أوباما على تعزيز الانقسام العنصري في أمريكا إن تصوير بيونسيه لنفسها كقديسة تقود النساء إلى المعمودية أمر مثير للضحك بالنظر إلى رسائل الكراهية والخلاف التي تواصل الترويج لها في أغانيها لقد أصبحت مجرد أداة لأوباما وحياة السود مهمة للجماعة الإرهابية أي شخص يشتري موسيقاها من شركة زوجها العنصرية فهو يساهم بشكل مباشر في الكراهية ظهرت والدا ترايفون مارتن ومايكل براون في الألبوم الجديد لبيونسيه، والذي تم إصداره على خدمة جاي زي تيدال للبث المباشر مساء السبت بعد عرض خاص على قناة للترويج له عصير الليمون هو عنوان الألبوم الأول للمغنية منذ ثلاث سنوات، والألبوم المرئي الذي تم بثه 

In [87]:
fd['Cleaned_Text'] = fd['Cleaned_Text'].str.replace('،', ' ', regex=False)

In [88]:
fd

y                                               text  \
0      0  beyonce doubles down…debuts #lemonade, another...   
1      1  mccain illness deprives senate of crucial vote...   
2      0  report: civil rights groups say voter suppress...   
3      0  disney worker tells horror story of being forc...   
4      1  trump 'weighed in' on son's russia attorney st...   
...   ..                                                ...   
42417  1  germany's schaeuble warns trump over protectio...   
42418  1  twenty-eight arrests after venezuela looting, ...   
42419  0  how the fbi cracked a terror plot on black fri...   
42420  1  u.s. tax plan threatens wall street's bermuda ...   
42421  1  asylum seekers refuse to leave papua new guine...   

                                                 text_ar  \
0      بيونسيه تتضاعف... لأول مرة #lemonade، أغنية أخ...   
1      مرض ماكين يحرم مجلس الشيوخ من التصويت الحاسم، ...   
2      تقرير: تقول جماعات الحقوق المدنية إن قمع الناخ...   
3      يروي عامل ديزني قصة مرعبة عن إجباره على تدريب ...   
4      البيت الأبيض: واشنطن (رويترز) - قال البيت الأب...   
...                                                  ...   
42417  وزير المالية الألماني فولفغانغ شيوبله يحذر ترا...   
42418  بورتو أورداز (فنزويلا) (رويترز) - قالت السلطات...   
42419  كيف تمكن مكتب التحقيقات الفيدرالي من كشف مؤامر...   
42420  نحن. خطة ضريبية تهدد منطقة برمودا في وول ستريت...   
42421  رفض مئات من طالبي اللجوء مغادرة مركز احتجاز في...   

                                            Cleaned_Text  
0      بيونسيه تتضاعف لأول مرة   أغنية أخرى لاستدراج ...  
1      مرض ماكين يحرم مجلس الشيوخ من التصويت الحاسم  ...  
2      تقرير تقول جماعات الحقوق المدنية إن قمع الناخب...  
3      يروي عامل ديزني قصة مرعبة عن إجباره على تدريب ...  
4      البيت الأبيض واشنطن رويترز قال البيت الأبيض يو...  
...                                                  ...  
42417  وزير المالية الألماني فولفغانغ شيوبله يحذر ترا...  
42418  بورتو أورداز فنزويلا رويترز قالت السلطات الفنز...  
42419  كيف تمكن مكتب التحقيقات الفيدرالي من كشف مؤامر...  
42420  نحن خطة ضريبية تهدد منطقة برمودا في وول ستريت ...  
42421  رفض مئات من طالبي اللجوء مغادرة مركز احتجاز في...  

[42422 rows x 4 columns]

In [89]:
def remove_extra_spaces(text):
    """
    تقوم باستبدال أي مسافة بيضاء متكررة (بما في ذلك التاب والأسطر الجديدة) بمسافة واحدة فقط.
    """
    # r'\s+' : يطابق أي تسلسل من المسافات البيضاء (سبيس، تاب، إلخ) مرة واحدة أو أكثر.
    cleaned_text = re.sub(r'\s+', ' ', text).strip()
    return cleaned_text

fd['Cleaned_Text'] = fd['Cleaned_Text'].apply(remove_extra_spaces)

In [90]:
fd

y                                               text  \
0      0  beyonce doubles down…debuts #lemonade, another...   
1      1  mccain illness deprives senate of crucial vote...   
2      0  report: civil rights groups say voter suppress...   
3      0  disney worker tells horror story of being forc...   
4      1  trump 'weighed in' on son's russia attorney st...   
...   ..                                                ...   
42417  1  germany's schaeuble warns trump over protectio...   
42418  1  twenty-eight arrests after venezuela looting, ...   
42419  0  how the fbi cracked a terror plot on black fri...   
42420  1  u.s. tax plan threatens wall street's bermuda ...   
42421  1  asylum seekers refuse to leave papua new guine...   

                                                 text_ar  \
0      بيونسيه تتضاعف... لأول مرة #lemonade، أغنية أخ...   
1      مرض ماكين يحرم مجلس الشيوخ من التصويت الحاسم، ...   
2      تقرير: تقول جماعات الحقوق المدنية إن قمع الناخ...   
3      يروي عامل ديزني قصة مرعبة عن إجباره على تدريب ...   
4      البيت الأبيض: واشنطن (رويترز) - قال البيت الأب...   
...                                                  ...   
42417  وزير المالية الألماني فولفغانغ شيوبله يحذر ترا...   
42418  بورتو أورداز (فنزويلا) (رويترز) - قالت السلطات...   
42419  كيف تمكن مكتب التحقيقات الفيدرالي من كشف مؤامر...   
42420  نحن. خطة ضريبية تهدد منطقة برمودا في وول ستريت...   
42421  رفض مئات من طالبي اللجوء مغادرة مركز احتجاز في...   

                                            Cleaned_Text  
0      بيونسيه تتضاعف لأول مرة أغنية أخرى لاستدراج ال...  
1      مرض ماكين يحرم مجلس الشيوخ من التصويت الحاسم و...  
2      تقرير تقول جماعات الحقوق المدنية إن قمع الناخب...  
3      يروي عامل ديزني قصة مرعبة عن إجباره على تدريب ...  
4      البيت الأبيض واشنطن رويترز قال البيت الأبيض يو...  
...                                                  ...  
42417  وزير المالية الألماني فولفغانغ شيوبله يحذر ترا...  
42418  بورتو أورداز فنزويلا رويترز قالت السلطات الفنز...  
42419  كيف تمكن مكتب التحقيقات الفيدرالي من كشف مؤامر...  
42420  نحن خطة ضريبية تهدد منطقة برمودا في وول ستريت ...  
42421  رفض مئات من طالبي اللجوء مغادرة مركز احتجاز في...  

[42422 rows x 4 columns]

In [91]:
fd = fd.drop(columns=["text_ar", "text"])

In [92]:
fd

y                                       Cleaned_Text
0      0  بيونسيه تتضاعف لأول مرة أغنية أخرى لاستدراج ال...
1      1  مرض ماكين يحرم مجلس الشيوخ من التصويت الحاسم و...
2      0  تقرير تقول جماعات الحقوق المدنية إن قمع الناخب...
3      0  يروي عامل ديزني قصة مرعبة عن إجباره على تدريب ...
4      1  البيت الأبيض واشنطن رويترز قال البيت الأبيض يو...
...   ..                                                ...
42417  1  وزير المالية الألماني فولفغانغ شيوبله يحذر ترا...
42418  1  بورتو أورداز فنزويلا رويترز قالت السلطات الفنز...
42419  0  كيف تمكن مكتب التحقيقات الفيدرالي من كشف مؤامر...
42420  1  نحن خطة ضريبية تهدد منطقة برمودا في وول ستريت ...
42421  1  رفض مئات من طالبي اللجوء مغادرة مركز احتجاز في...

[42422 rows x 2 columns]

In [93]:
dfc.to_csv('arabic_clear.csv', index=False, encoding='utf-8-sig')

In [94]:
import nltk
from nltk.corpus import stopwords

# تحميل قائمة كلمات التوقف العربية
try:
    stopwords.words('arabic')
except LookupError:
    nltk.download('stopwords')

arabic_stopwords = set(stopwords.words('arabic'))

# تعريف الدالة التي تنظف النص (كما في الشرح السابق)
def clean_arabic_text(text):
    if not isinstance(text, str): # تأكد من أن القيمة نص قبل المعالجة
        return text

    words = text.split()
    filtered_words = []

    for word in words:
        # أ. تنظيف الكلمة من علامات الترقيم
        word = re.sub(r'[^\w\s]', '', word)

        # ب. الإزالة الأساسية لكلمات التوقف
        if word not in arabic_stopwords and len(word) > 1:
            # ج. معالجة الـ التعريف وحروف الجر الملتصقة في بداية الكلمة
            stemmed_word = word
            if word.startswith('ال'):
                stemmed_word = word[2:] # حذف "ال"
            elif word.startswith('و'):
                stemmed_word = word[1:] # حذف "و" الملتصقة

            # د. إعادة فحص الكلمة بعد التعديل
            if stemmed_word not in arabic_stopwords and stemmed_word != 'ب':
                filtered_words.append(stemmed_word)

    return ' '.join(filtered_words)

In [95]:
fd['text'] = fd['Cleaned_Text'].apply(clean_arabic_text)

In [96]:
fd

y                                       Cleaned_Text  \
0      0  بيونسيه تتضاعف لأول مرة أغنية أخرى لاستدراج ال...   
1      1  مرض ماكين يحرم مجلس الشيوخ من التصويت الحاسم و...   
2      0  تقرير تقول جماعات الحقوق المدنية إن قمع الناخب...   
3      0  يروي عامل ديزني قصة مرعبة عن إجباره على تدريب ...   
4      1  البيت الأبيض واشنطن رويترز قال البيت الأبيض يو...   
...   ..                                                ...   
42417  1  وزير المالية الألماني فولفغانغ شيوبله يحذر ترا...   
42418  1  بورتو أورداز فنزويلا رويترز قالت السلطات الفنز...   
42419  0  كيف تمكن مكتب التحقيقات الفيدرالي من كشف مؤامر...   
42420  1  نحن خطة ضريبية تهدد منطقة برمودا في وول ستريت ...   
42421  1  رفض مئات من طالبي اللجوء مغادرة مركز احتجاز في...   

                                                    text  
0      بيونسيه تتضاعف لأول مرة أغنية أخرى لاستدراج عر...  
1      مرض ماكين يحرم مجلس شيوخ تصويت حاسم اشنطن رويت...  
2      تقرير تقول جماعات حقوق مدنية قمع ناخبين مسؤول ...  
3      يروي عامل ديزني قصة مرعبة إجباره تدريب عامل أج...  
4      بيت أبيض اشنطن رويترز قال بيت أبيض يوم ترامب ب...  
...                                                  ...  
42417  زير مالية ألماني فولفغانغ شيوبله يحذر ترامب حم...  
42418  بورتو أورداز فنزويلا رويترز قالت سلطات فنزويلي...  
42419  تمكن مكتب تحقيقات فيدرالي كشف مؤامرة إرهابية ي...  
42420  خطة ضريبية تهدد منطقة برمودا ول ستريت نيويورك ...  
42421  رفض مئات طالبي لجوء مغادرة مركز احتجاز بابوا غ...  

[42422 rows x 3 columns]

In [97]:
fd = fd.drop(columns=['Cleaned_Text'])

In [98]:
fd

y                                               text
0      0  بيونسيه تتضاعف لأول مرة أغنية أخرى لاستدراج عر...
1      1  مرض ماكين يحرم مجلس شيوخ تصويت حاسم اشنطن رويت...
2      0  تقرير تقول جماعات حقوق مدنية قمع ناخبين مسؤول ...
3      0  يروي عامل ديزني قصة مرعبة إجباره تدريب عامل أج...
4      1  بيت أبيض اشنطن رويترز قال بيت أبيض يوم ترامب ب...
...   ..                                                ...
42417  1  زير مالية ألماني فولفغانغ شيوبله يحذر ترامب حم...
42418  1  بورتو أورداز فنزويلا رويترز قالت سلطات فنزويلي...
42419  0  تمكن مكتب تحقيقات فيدرالي كشف مؤامرة إرهابية ي...
42420  1  خطة ضريبية تهدد منطقة برمودا ول ستريت نيويورك ...
42421  1  رفض مئات طالبي لجوء مغادرة مركز احتجاز بابوا غ...

[42422 rows x 2 columns]

In [99]:
fd.to_csv('arabic_f.csv', index=False, encoding='utf-8-sig')